bpf/cpu/cpu.bpf.c

```diff
@@ -622,22 +622,12 @@ static __always_inline bool has_fp(u64 current_fp) {
 static __always_inline void add_stack(struct bpf_perf_event_data *ctx, u64 pid_tgid, enum stack_walking_method method, unwind_state_t *unwind_state) {
   stack_count_key_t *stack_key = &unwind_state->stack_key;
 
-  // The `bpf_get_current_pid_tgid` helpers returns
-  // `current_task->tgid << 32 | current_task->pid`, the naming can be
-  // confusing because the thread group identifier and process identifier
-  // mean different things in kernel and user space.
-  //
-  // - What we call PIDs in userspace, are TGIDs in kernel space.
-  // - What we call **thread IDs** in user space, are PIDs in kernel space.
-  // In other words, the process ID in the lower 32 bits (kernel's view of the PID,
-  // which in user space is usually presented as the thread ID),
-  // and the thread group ID in the upper 32 bits
-  // (what user space often thinks of as the PID).
-
-  int user_pid = pid_tgid >> 32;
-  int user_tgid = pid_tgid;
-  stack_key->pid = user_pid;
-  stack_key->tgid = user_tgid;
+
+  int per_process_id = pid_tgid >> 32;
+  int per_thread_id = pid_tgid;
```

Comment:
I find tgid and pid confusing because they are context dependent, these names mean what it says on the tin: any thread in a process should return the same `per_process_id` while each thread should have a unique `per_thread_id`

Reply:
Let's have this as an inline code comment. WDYT?

Reply:
We encountered a few misunderstandings around this piece of the code. I'm in for anything that helps us to prevent future mistakes. Unfortunately, I don't have any valuable suggestions besides inline comments.

Reply:
I don't mind adding the comments but I just think it will be difficult reading the code and then we need to ensure that they will be kept in sync in 2-3 different places